gen3utils/gitops/gitops_validator.py

```diff
@@ -0,0 +1,385 @@
+                    datatype
+                )
+            )
+        )
+
+    tabs = explorer_config["filters"]["tabs"]
+    for tab in tabs:
+        errors = check_field_value(
+            "explorerConfig.filters.tabs.fields", tab.get("fields", []), props, errors
+        )
+
+    table = explorer_config["table"]
+    if table["enabled"]:
+        errors = check_field_value(
+            "explorerConfig.table.fields", table.get("fields", []), props, errors
+        )
+
+    manifest_map = explorer_config.get("manifestMapping")
+    if manifest_map and manifest_map.get("resourceIndexType"):
+        resource_props = type_prop_map.get(manifest_map.get("resourceIndexType"))
+        if not resource_props:
+            errors.append(
+                FieldError(
+                    "Invalid field {} in manifestMapping.resourceIndexType".format(
+                        manifest_map.get("resourceIndexType")
+                    )
+                )
+            )
+        elif manifest_map.get("resourceIdField") not in resource_props:
+            errors.append(
+                FieldError(
+                    "Invalid field {} in manifestMapping.resourceIdField".format(
+                        manifest_map.get("resourceIdField")
+                    )
+                )
+            )
+        # TODO
+        # Also consider fields referenceIdFieldInResourceIndex and referenceIdFieldInDataIndex
+
+    return errors
+
+
+def validate_explorerConfig(gitops, type_prop_map, errors):
+    exploreConfig = gitops.get("explorerConfig") or gitops.get("dataExplorerConfig")
+    # if explorerConfig exists, ignores (data/files)explorerConfig
+    if exploreConfig and type(exploreConfig) == list:
+        for config in exploreConfig:
+            errors = _validate_explorerConfig_helper(config, type_prop_map, errors)
+    else:
+        errors = _validate_explorerConfig_helper(exploreConfig, type_prop_map, errors)
+        file_exp_config = gitops.get("fileExplorerConfig")
+        if file_exp_config:
+            errors = _validate_explorerConfig_helper(
+                file_exp_config, type_prop_map, errors
+            )
+
+    return errors
+
+
+def validate_against_etl(gitops, mapping_file):
+    """
+    Validates gitops.json configuration against an etlMapping
+    Args:
+        gitops (dict): gitops.json config
+        mapping_file (str): path to etlMapping file
+
+    Returns:
+        Error: Returns a list of any errors encountered.
+
+    """
+    with open(mapping_file) as f:
+        mappings = yaml.safe_load(f)
+    mapping = mappings.get("mappings")
+    type_prop_map = map_all_ES_index_props(mapping)
+    errors = validate_explorerConfig(gitops, type_prop_map, [])
+    studyviewer = gitops.get("studyViewerConfig")
+    if studyviewer:
+        errors = validate_studyViewerConfig(studyviewer, type_prop_map, errors)
+
+    return errors
+
+
+def map_all_ES_index_props(mapping):
```

Comment:
i think we should take the index type ("aggregator" or "collector") into account. For example, props in `aggregated_props` will be included in the index if the index type is `aggregator`, but not if it's `collector`

Reply:
out of scope; created ticket PXP-7863 to add this to ETL mapping validation